### PR TITLE
OPENEUROPA-754: Parameters in pCAS service file to be easy to be overridden

### PR DESCRIPTION
**Pull request by aritomelo**
- Create:
  New service that encapsulates all pcas settings (allowing to pass the service to the pCAS constructor instead of a set of config)
All config entry has been flattening so it can be easier to override them (can be selectively overridden)

### CODE REVIEW

File: Resources/config/p_cas.yml [hunk 1]
```diff
@@ -1,39 +1,42 @@
...
-      logout:
-        uri: http://127.0.0.1:8000/logout
-        query: []
-        allowed_parameters:
-          - service
+  logger_startup_message: 'Logger just got started!!! (from original config)'
+
+  protocol.login.uri: http://127.0.0.1:8000/login
+  protocol.login.query: []
+  protocol.login.allowed_parameters:
```
Comment (pfrenssen) [inline Resources/config/p_cas.yml]: This seems like a step back. It was better before. This configuration has a clear structure, so I would not change it.

**Comment by pfrenssen** [comment]
I would not store the settings in a service. There is no need to have the settings for this component available on the dependency injection container as a singleton.

I don't know any current framework that uses this approach of using a singleton to store fixed config.

* Drupal: stores config in YAML files and has the config schema defined in another YAML file.
* Symfony: stores config in YAML files and has the config schema defined in a static PHP method.
* Laravel: stores config in .env files.

Let's keep it in YAML, this is very easy to override already.

File: Resources/config/p_cas.yml (hunk 1 above)
Comment (aritomelo) [reply Resources/config/p_cas.yml]: @pfrenssen It would be good if you could also add your comments on the ticket#754, please.

**Comment by drupol** [comment]
@pfrenssen, I'm sharing the same thoughts.

I don't see any issue overriding an array.

If building the array is an issue, we could implement a static method like:

```php
$config = PCas::getDefaultConfig();
// Modify the config
$config['foo'] = 'bar';
$pcas = new PCas($config);
```

File: src/Cas/Protocol/V2/CasProtocolV2.php [hunk 2]
```diff
@@ -48,39 +48,6 @@ public function validateServiceTicket($serviceTicket, $requestPgt = false, $url
             }
             $pCasUser = $validatedResponse;
 
-
```
Comment (drupol) [inline src/Cas/Protocol/V2/CasProtocolV2.php]: Why removing this ? This is not part of the PR scope I think

File: demo-server/symfony.lock [hunk 3]
```diff
@@ -17,12 +17,21 @@
     "doctrine/common": {
         "version": "v2.8.1"
     },
+    "doctrine/event-manager": {
```
Comment (drupol) [inline demo-server/symfony.lock]: Why adding this ? Is this needed now ?

File: src/Cas/Protocol/V2/CasProtocolV2.php (hunk 2 above)
Comment (aritomelo) [reply src/Cas/Protocol/V2/CasProtocolV2.php]: This was commented out (not in use), so just for clean up!

**Comment by ademarco** [comment]
After discussing it with @aritomelo  and @imanoleguskiza  we concluded that the factory object being worked in OPENEUROPA-755 will ensure configuration to be [easily override-able](https://github.com/openeuropa/pcas/pull/18#discussion_r208168673). That will also allow too keep the current pCAS behavior intact, which will be convenient for those already using the library.

Closing as no action is required here anymore.
